pinc/misc.inc

```diff
@@ -882,14 +882,30 @@ function flatten_directory(string $directory_to_flatten)
  * If any of the conditions are not satisfied, it will return false, otherwise it will return true.
  * The extension check can be disabled by passing true as the second argument.
  */
+
+class ZipException extends Exception
+{
+}
+
 function is_valid_zip_file(string $file_path, bool $ignore_extension_check = false): bool
 {
-    if (!file_exists($file_path)) {
+    try {
+        validate_zip_file($file_path, $ignore_extension_check);
+    } catch (ZipException $exception) {
+        error_log("is_valid_zip_file: " . $exception->getMessage());
```

Comment:
Our usual `error_log()` format string in the code is `<filename> - $error` but that doesn't make a lot of sense here as `misc.inc` is a common include. Consider instead: `is_valid_zip_file() - $error`.

In general I'm a little dubious of logging an error here in a generic validation function. I suspect as a pattern that will generate a fair amount of noise compared to the value. It might be better to have this function just return true/false as it did before without logging the exception and update `zip_check()` to run the new validation function and capture and log the error there with the filename in the message like we do elsewhere.

Reply:
I'll change it to work like that.